fix: CI disk space exhaustion — install_smoke runs twice, fills runner

## Problem
The `pre-commit` CI job runs `cargo test --all-features --locked` (pre-push stage), which includes `tests/install_smoke.rs`. This test does `cargo install --path .` in **release mode**, compiling the full binary a second time after debug targets are already built. The runner runs out of disk space compiling the `ring` crate's C code:

```
ring-0.17.14/crypto/curve25519/curve25519.c:1924:1: fatal error: error writing to /tmp/ccJWuGeI.s: No space left on device
```

This is **blocking all PRs**, including:
- #325 (improvement_curation test fix)
- #294 (test round 5: +1,740 lines)
- #321 (test round 6: +1,559 lines)

## Root Cause
`install_smoke` already has its own dedicated `install-smoke` job in `verify.yml` which succeeds. The pre-push hook's `cargo test` duplicates this and the combined debug+release builds exceed runner disk.

## Fix
Exclude `install_smoke` from the pre-push hook's `cargo test` by adding `--exclude` or using `--skip` filter. The dedicated `install-smoke` job handles it.

Options:
1. Change pre-commit hook entry to: `cargo test --all-features --locked -- --skip cargo_install_from_repo_succeeds`
2. Or move install_smoke.rs to a separate crate/test target excluded from default test

## Impact
Unblocks **3,300+ lines of test code** across PRs #294 and #321, directly advancing the test coverage goal.

Relates to #327

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -27,8 +27,8 @@ repos:
           - manual
 
       - id: cargo-test
-        name: cargo test --all-features --locked
-        entry: cargo test --all-features --locked
+        name: cargo test --all-features --locked (skip install_smoke)
+        entry: cargo test --all-features --locked -- --skip cargo_install_from_repo_succeeds
         language: system
         pass_filenames: false
         always_run: true

